File: app/src/main/java/com/knu/krasn/knuscheduler/Presenter/Network/ApiInterface.java
package com.knu.krasn.knuscheduler.Presenter.Network;


import com.knu.krasn.knuscheduler.Model.Models.Pojos.Faculties.Faculties;
import com.knu.krasn.knuscheduler.Model.Models.Pojos.GroupModel.Groups;
import com.knu.krasn.knuscheduler.Model.Models.Pojos.Schedule.Schedules;

import java.util.Map;

import io.reactivex.Single;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.HeaderMap;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Created by krasn on 10/29/2017.
 */

public interface ApiInterface {

    @GET("getFacultiesList")
    Single<Faculties> getFaculties(@HeaderMap Map<String, String> headerMap);

    @GET("getGroupsList")
    Single<Groups> getGroups(@Query("faculty_id") String facultyID, @HeaderMap Map<String, String> headerMap);

    @FormUrlEncoded
    @POST("getSchedule")
    Single<Schedules> getSchedule(@Field("group") String group, @HeaderMap Map<String, String> headerMap);

    @GET("getAdvanceSchedule")
    Single<Schedules> getSearchingSchedule(@Query("query") String searchQuery, @HeaderMap Map<String, String> headerMap);
}

File: app/src/main/java/com/knu/krasn/knuscheduler/Repository/Network/ApiInterface.kt
package com.knu.krasn.knuscheduler.Repository.Network


import com.knu.krasn.knuscheduler.Repository.FacultyResponse
import com.knu.krasn.knuscheduler.Repository.GroupResponse
import com.knu.krasn.knuscheduler.Repository.Schedule
import kotlinx.coroutines.experimental.Deferred
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST

interface ApiInterface {

    @GET("getFacultiesList")
    fun getFaculties():Deferred<FacultyResponse>
    @GET("getGroupsList")
fun getGroups():Deferred<GroupResponse>
    @FormUrlEncoded
    @POST("getSchedule")
fun getSchedule():Deferred<Schedule>

    @GET("getAdvanceSchedule")
fun getAdvanceSchedule():Deferred<Schedule>
}